File: leetcode/algorithms/easy/remove-duplicates-from-sorted-list.js
/*
    leetcode
    category: "Algorithms"
    problem title: "Remove Duplicates From Sorted List"
*/

/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    if(!head){
        return head;
    }
    let trav = head;
    while(trav.next!=null){
        if(trav.next.val===trav.val){
            trav.next = trav.next.next;
        } else {
            trav = trav.next;
        }
    }
    return head;
    
};

File: leetcode/algorithms/linked-list/remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        
        ListNode curr = head;
    
        while(curr!=null){
            if(curr.next!=null && curr.val==curr.next.val){
                ListNode nxtNode = curr.next;
                while(nxtNode!=null && nxtNode.val==curr.val){
                    nxtNode = nxtNode.next;
                }
                curr.next = nxtNode;
            } else {
                curr = curr.next;
            }
        }
        
        return head;
        
    }
}